docs/getting-started.md

```diff
@@ -0,0 +1,325 @@
+# Getting started with JupyterHub
+
+This document describes some of the basics of configuring JupyterHub to do what you want.
+JupyterHub is highly customizable, so there's a lot to cover.
+
+
+## Installation
+
+See [the readme](../README.md) for help installing JupyterHub.
+
+
+## JupyterHub's default behavior
+
+Let's start by describing what happens when you type `sudo jupyterhub`
+after installing it, without any configuration.
+
+
+### Authentication
+
+The default Authenticator that ships with JupyterHub
+authenticates users with their system name and password (via [PAM][]).
+Any user on the system with a password will be allowed to start a notebook server.
+
+
+### Spawning servers
+
+The default Spawner starts servers locally as each user,
+one for each server. These servers listen on localhost,
+and start in the given user's home directory.
+
+
+### Network
+
+JupterHub consists of three main categories of processes:
+
+- Proxy
+- Hub
+- Spawners
+
+The Proxy is the public face of the service.
+Users access the server via the proxy.
+By default, this is listening on all public interfaces on port 8000.
+You can access the hub at:
+
+    http://localhost:8000
+
+or any other IP or domain pointing for your system.
+
+The other services, Hub and Spawners, all communicate with each other on localhost only.
+If you are going to separate these processes across machines or containers,
+you may need to tell them to listen on addresses other than localhost.
+
+**NOTE** this server is running without SSL encryption.
+You should not run JupyterHub without HTTPS if you can help it.
+
+
+### Files
+
+Starting JupyterHub will write two files to disk in the current working directory:
+
+- `jupyterhub.sqlite` is the sqlite database containing all of the state of the Hub.
+  This file allows the Hub to remember what users are running and where,
+  as well as other information enabling you to 
+  You can change the location of this file with `--db=/path/to/somedb.sqlite`.
+- `jupyterhub_cookie_secret` is the encryption key used for securing cookies.
+  This file needs to persist in order for restarting the Hub server to avoid invalidating cookies.
+  Conversely, deleting this file and restarting the server effectively invalidates all login cookies.
+
+
+## How to configure JupyterHub
+
+JupyterHub is configured in two ways:
+
+- command-line arguments. see `jupyterhub -h` for information about the arguments,
+  or `jupyterhub --help-all` for a list of everything configurable on the command-line.
+- config files. The default config file is `jupyterhub_config.py`, in the current working directory.
+  You can create an empty config file with `jupyterhub --generate-config`
+  to see all the configurable values.
+  You can load a specific config file with `jupyterhub -f /path/to/jupyterhub_config.py`.
+
+
+## Networking
+
+When it starts, JupyterHub creates two processes:
```

Comment:
It seems like this section is just repeating the information that is above in the default configuration section. I think it may make sense to combine these two overall section (## default and ## configuring) and then start each sub-section by describing what the default setup is.